Added right fixed columns support

wenzhixin opened the pull request:
* Fix #2266
* Fix #3480
* Fix #4510
* Fix #4534
* Fix #4584
* Fix #4587
* Fix #4649
* Fix #4775
* Fix #4797

Examples:

* Left and right without `height`: https://live.bootstrap-table.com/code/wenzhixin/1714
* Left and right with `height`: https://live.bootstrap-table.com/code/wenzhixin/1711
* Checkbox: https://live.bootstrap-table.com/code/wenzhixin/1713
* Auto enable fixed columns: https://live.bootstrap-table.com/code/wenzhixin/1712

djhvscf commented:
Hi @wenzhixin  there are some css problems when height is set

djhvscf commented on src/extensions/fixed-columns/bootstrap-table-fixed-columns.js:
```diff
@@ -56,58 +51,244 @@ $.BootstrapTable = class extends $.BootstrapTable {
+    }
+
+    this.$tableBody.find('tr').each((i, el) => {
+      const index = $(el).data('index')
```
Can we save the $(el) in a const variable and use it in the next lines?

djhvscf commented on src/extensions/fixed-columns/bootstrap-table-fixed-columns.js:
```diff
@@ -56,58 +51,244 @@ $.BootstrapTable = class extends $.BootstrapTable {
+    }
+
+    if (this.needFixedColumns && this.options.fixedNumber) {
+      this.$fixedColumns.find('.fixed-table-header').remove()
```
Why removing .fixed-table-header and then find it again?

UtechtDustin replied:
The header will be recreated, thats the reason i guess.

UtechtDustin replied:
This 3 lines will be used often, maybe create a function for that ?

djhvscf commented on src/extensions/fixed-columns/bootstrap-table-fixed-columns.js:
```diff
@@ -56,58 +51,244 @@ $.BootstrapTable = class extends $.BootstrapTable {
+    }
+
+    if (this.needFixedColumns && this.options.fixedRightNumber) {
+      this.$fixedColumnsRight.find('.fixed-table-header').remove()
```
Same here

djhvscf commented on src/extensions/fixed-columns/bootstrap-table-fixed-columns.js:
```diff
@@ -56,58 +51,244 @@ $.BootstrapTable = class extends $.BootstrapTable {
+          this.$fixedBodyRight.scrollTop(top)
+        }
+      }).find('tr').hover(e => {
+        const index = $(e.currentTarget).data('index')
```
Can we save the e.currentTarget in a const variable and use it in next lines?

wenzhixin commented:
Finished this PR.

> there are some css problems when height is set

What problems?

UtechtDustin commented:
Is ist compatible with all extensions ?

wenzhixin commented:
@djhvscf windows + chrome?

djhvscf commented:
Yes, sir

UtechtDustin commented:
@wenzhixin 
Same for Linux + Firefox

djhvscf commented:
If this helps

Windows + New Microsoft Edge

UtechtDustin commented:
Please also check if this PR maybe fix #4510

wenzhixin commented:
Finished, but the online example cannot work because of the cache.

wenzhixin commented:
Fixed the bug of firefox and added all themes support:

* Bootstrap v3: https://live.bootstrap-table.com/code/wenzhixin/1715
* Bulma: https://live.bootstrap-table.com/code/wenzhixin/1716
* Semantic: https://live.bootstrap-table.com/code/wenzhixin/1717
* Materialize: https://live.bootstrap-table.com/code/wenzhixin/1718
* Foundation: https://live.bootstrap-table.com/code/wenzhixin/1719

UtechtDustin commented:
Could you please update the examples int the first comment ? :-)

UtechtDustin commented:
Looks like #4649 is also fixed with that PR, could you @wenzhixin please verify that ?

wenzhixin commented:
> Could you please update the examples int the first comment ? :-)

updated

djhvscf commented:
> > Could you please update the examples int the first comment ? :-)
> 
> updated

No more issues with css. Approved for my point of view

Thanks @wenzhixin

wenzhixin commented:
> Looks like #4649 is also fixed with that PR, could you @wenzhixin please verify that ?

Yes.

UtechtDustin commented:
I can also approve that PR, thank your @wenzhixin for the great PR :-)
